kernel/timer.c

```diff
@@ -126,7 +126,9 @@ void z_impl_k_timer_start(struct k_timer *timer, k_timeout_t duration,
 	 * argument the same way k_sleep() does), but historical.  The
 	 * timer_api test relies on this behavior.
 	 */
-	period.ticks = MAX(period.ticks - 1, 0);
+	if (period.ticks != 0 && Z_TICK_ABS(period.ticks) >= 0) {
```

Comment:
`Z_TICK_ABS()` returns non-negative when the input value is an absolute timeout, so this adjusts the value.

This seems to be inconsistent with the commit description which says it's not appropriate to adjust an absolute timeout period.  (Though such a thing should probably be documented as undefined, and we could just ignore it.)

What am I missing?

Reply:
Nothing, it was wrong.